noobaa_sa/s3_client.py

```diff
@@ -145,21 +209,37 @@ def list_objects(self, bucket_name, prefix="", use_v2=False):
             else self._boto3_client.list_objects
         )
         output = list_objects_method(Bucket=bucket_name, Prefix=prefix)
-        listed_objects = []
+        listed_obj_md_dicts = []
         if "Contents" in output:
-            listed_objects = [obj["Key"] for obj in output["Contents"]]
-        log.info(f"Listed objects: {listed_objects}")
-        return listed_objects
+            listed_obj_md_dicts = output["Contents"]
+        listed_objs_names = [obj["Key"] for obj in listed_obj_md_dicts]
+        log.info(f"Listed objects: {listed_objs_names}")
+        return listed_obj_md_dicts if get_metadata else listed_objs_names
 
-    def put_object(self, bucket_name, object_key, object_data):
+    def put_object(self, bucket_name, object_key, body):
         """
-        Put an object in an S3 bucket using boto3
+        Put an object to an S3 bucket using boto3
+
+        Args:
+            bucket_name (str): The name of the bucket
+            object_key (str): The key of the object
+            body (bytes|file-like object): The data to write to the object
+
+        Raises:
+            NoSuchBucket: If the bucket does not exist
 
         """
         log.info(f"Putting object {object_key} in bucket {bucket_name} via boto3")
-        self._boto3_client.put_object(
-            Bucket=bucket_name, Key=object_key, Body=object_data
-        )
+        try:
+            self._boto3_client.put_object(Bucket=bucket_name, Key=object_key, Body=body)
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchBucket":
+                log.warn(
+                    f"Bucket {bucket_name} does not exist and cannot be written to"
+                )
+                raise NoSuchBucket(e)
+            else:
+                raise e
```

Comment:
There might be situation where we need to verify the operations based on response code. I would recommend here to return response dictionary for these boto3 methods

Reply:
Done in https://github.com/red-hat-storage/noobaa-sa-ci/pull/16/commits/cde85fc0c8aae17993097f4a72b721a00cc42873